src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,369 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.decomposition.signed_hahn
+
+/-!
+# Jordan decomposition
+
+This file proves the existence and uniqueness of the Jordan decomposition for signed measures.
+The Jordan decomposition theorem states that, given a signed measure `s`, there exists a
+unique pair of mutually singular measures `μ` and `ν`, such that `s = μ - ν`.
+
+The Jordan decomposition theorem for measures is a corollary of the Hahn decomposition theorem and
+is useful for the Lebesgue decomposition theorem.
+
+## Main definitions
+
+* `measure_theory.jordan_decomposition`: a Jordan decomposition of a measurable space is a
+  pair of mutually singular finite measures. We say `j` is a Jordan decomposition of a signed
+  meausre `s` if `s = j.μ - j.ν`.
+* `measure_theory.signed_measure.to_jordan_decomposition`: the Jordan decomposition of a
+  signed measure.
+* `measure_theory.signed_measure.to_jordan_decomposition_equiv`: is the `equiv` between
+  `measure_theory.signed_measure` and `measure_theory.jordan_decomposition` formed by
+  `measure_theory.signed_measure.to_jordan_decomposition`.
+
+## Main results
+
+* `measure_theory.signed_measure.to_signed_measure_to_jordan_decomposition` : the Jordan
+  decomposition theorem.
+* `measure_thoery.signed_measure.to_signed_measure_injective` : the Jordan decomposition of a
+  signed measure is unique.
+
+## Tags
+
+Jordan decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory
+
+variables {α β : Type*} [measurable_space α]
+
+namespace measure_theory
+
+/-- A Jordan decomposition of a measurable space is a pair of mutually singular,
+finite measures. -/
+@[ext] structure jordan_decomposition (α : Type*) [measurable_space α] :=
+(μ ν : measure α)
```

Comment:
I don't like mu and nu here, because it is impossible to guess which is which. What about `pos_part` and `neg_part`, just like we have for functions.